beacon-chain/cache/checkpoint_state.go

```diff
@@ -99,7 +100,10 @@ func (c *CheckpointStateCache) StateByCheckpoint(cp *ethpb.Checkpoint) (*stateTr
 func (c *CheckpointStateCache) AddCheckpointState(cp *CheckpointState) error {
 	c.lock.Lock()
 	defer c.lock.Unlock()
-	if err := c.cache.AddIfNotPresent(cp); err != nil {
+	if err := c.cache.AddIfNotPresent(&CheckpointState{
+		Checkpoint: proto.Clone(cp.Checkpoint).(*ethpb.Checkpoint),
```

Comment:
Lets use our custom copy over here instead of proto clone

Reply:
Done